### ISSUE TITLE
Disambiguate types

### ISSUE THREAD

**Issue by reynir** [ocaml-semver/ocaml-api-watch#84]
I am writing this issue based off of the Outreachy demo earlier this year. Please correct me if the following is not the case anymore.

Inside a nested module `Foo.Bar.Baz` there might be a value `doit` with the following type:

```OCaml
val doit : t -> unit
```

In a diff without context it's unclear if `t` refers to `Foo.Bar.Baz.t`, `Foo.Bar.t` or `Foo.t`. I think in error messages I have seen `t/1` and `t/2` to precise what `t` is referred to. Alternatively, the type could be fully qualified, e.g. `val doit : Foo.Bar.t -> unit`.

**Comment by NathanReb** [comment]
Yes you're right that this is important and definitely something we'll be aiming for in the output of the diff.

There is still quite a bit of work to do to properly set the typing environment for our diffing needs as we went with the bare minimum for the first prototype. For example it will fail if your API refers to types defined by another library, external to the one you're diffing.

Once we properly set the typing env, always using the fully qualified name seems like a good idea. For the sake of readibality we could also consider reserving non qualified names for types that are defined in the module itself.
